src/Middleware/ImplicitHeadMiddlewareFactory.php

```diff
@@ -13,32 +13,24 @@
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\StreamInterface;
 use Zend\Expressive\Router\Exception\MissingDependencyException;
+use Zend\Expressive\Router\RouterInterface;
 
 /**
  * Create and return an ImplicitHeadMiddleware instance.
  *
  * This factory depends on two other services:
  *
- * - Psr\Http\Message\ResponseInterface, which should resolve to a callable
- *   that will produce an empty Psr\Http\Message\ResponseInterface instance.
  * - Psr\Http\Message\StreamInterface, which should resolve to a callable
  *   that will produce an empty Psr\Http\Message\StreamInterface instance.
  */
 class ImplicitHeadMiddlewareFactory
 {
     /**
-     * @throws MissingDependencyException if either the Psr\Http\Message\ResponseInterface
-     *     or Psr\Http\Message\StreamInterface services are missing.
+     * @throws MissingDependencyException if the Psr\Http\Message\StreamInterface
+     *     service is missing.
      */
     public function __invoke(ContainerInterface $container) : ImplicitHeadMiddleware
     {
-        if (! $container->has(ResponseInterface::class)) {
-            throw MissingDependencyException::dependencyForService(
-                ResponseInterface::class,
-                ImplicitHeadMiddleware::class
-            );
-        }
-
```

Comment:
Why is this removed, but the one throwing for a stream interface is not? We should either check for both, or neither.

Reply:
Never mind; I see now it was testing for a response when the instance never receives one.